.github/workflows/build.yml

```
@@ -0,0 +1,243 @@
+            echo "No branch specified, using workflow branch: $WORKFLOW_BRANCH"
+            RESOLVED_BRANCH="$WORKFLOW_BRANCH"
+          else
+            echo "Using specified branch: $INPUT_BRANCH"
+            RESOLVED_BRANCH="$INPUT_BRANCH"
+          fi
+
+          echo "branch=$RESOLVED_BRANCH" >> "$GITHUB_OUTPUT"
+          echo "RESOLVED_BRANCH=$RESOLVED_BRANCH" >> "$GITHUB_ENV"
+
+      - name: Checkout Code
+        uses: actions/checkout@v3
```

actionlint (1.7.10)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)